File: gdb/domain/fixed_deposit_account.py

from gdb.domain.abstract_account import AbstractAccount
from gdb.exceptions import AccountException

class FixedDepositAccount(AbstractAccount):
    def __init__(self, account_number, name, age, balance, status, pin, tenure_months, interest_rate):
        super().__init__(account_number, name, age, balance, "FIXEDDEPOSIT", status, pin)
        
    def process_debit(self, amount: float) -> None:
        raise AccountException("Premature withdrawals are not permitted on Fixed Deposit accounts")

File: src/com/gdb/domain/FixedDepositAccount.java
package com.gdb.domain;
import com.gdb.exceptions.*;

public class FixedDepositAccount extends AbstractAccount {
    private int tenureMonths;
    private double interestRate;

    public FixedDepositAccount(String accountNumber, String name, int age, double balance, String status, String pin, int tenureMonths, double interestRate) {
        super(accountNumber, name, age, balance, "FIXED_DEPOSIT", status, pin);
        this.tenureMonths = tenureMonths;
        this.interestRate = interestRate;
    }

    public double calculateMaturityAmount() {
        return this.balance * Math.pow(1 + (interestRate / 100.0), tenureMonths / 12.0);
    }

    @Override
    protected void processDebit(double amount) throws AccountException {
        throw new AccountException("Premature withdrawals are not permitted on Fixed Deposit accounts");
    }
}
